supervisor/api/store.py

```diff
@@ -136,6 +137,8 @@ def addons_addon_install(self, request: web.Request) -> Awaitable[None]:
     def addons_addon_update(self, request: web.Request) -> Awaitable[None]:
         """Update add-on."""
         addon = self._extract_addon(request, installed=True)
+        if addon == request.get(REQUEST_FROM):
```

Comment:
This is already blocked, right?
<https://github.com/home-assistant/supervisor/blob/main/supervisor/api/middleware_security.py#L53>

Reply:
The self is not working on an update with the new store API and you can just not use `self` and it works. The Store idea was like on android, the apps can also not doing self-update i.e. to load new code without user knowlage